core/commands/swarm.go

```diff
@@ -128,7 +129,11 @@ var swarmPeersCmd = &cmds.Command{
 
 			buf := new(bytes.Buffer)
 			for _, info := range ci.Peers {
-				fmt.Fprintf(buf, "%s/ipfs/%s", info.Addr, info.Peer)
+				if strings.HasSuffix(info.Addr, fmt.Sprintf("/ipfs/%s", info.Peer)) {
+					fmt.Fprintf(buf, "%s", info.Addr)
+				} else {
+					fmt.Fprintf(buf, "%s/ipfs/%s", info.Addr, info.Peer)
```

Comment:
I can certainly change it to use the scheme name from multiaddr. As for multiaddr method manipulation, these are strings here so that's inconvenient.

And yes indeed, I am guilty of using hardcoded ipfs strings in go-libp2p-circuit. I will look into fixing it there as well, although it's not critical for this PR.